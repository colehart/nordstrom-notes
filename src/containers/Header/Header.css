.Header {
  grid-row: 1 / 2;
  grid-column: 1 / 2;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.hidden {
  display: none;
}

.btn {
  height: 37px;
  display: flex;
  justify-content: center;
  align-items: flex-end;
  border-bottom: 2px solid transparent;
  padding-bottom: 3px;
  text-decoration: none;
  font-size: 1.15rem;
  color: black;
  transition: 0.2s;
}

.btn:hover {
  border-bottom-color: var(--medium-gray);
}

.btn:active {
  border-bottom-color: var(--gray);
}

.h-logo img {
  width: 350px;
  transition: 0.2s;
}

.h-logo:active {
  opacity: 0.7;
}

.h-nav-group {
  display: flex;
  justify-content: space-between;
  align-items: center;
  text-decoration: none;
  color: white;
  height: 200px;
  width: 300px;
}

.h-error {
  position: absolute;
  top: 0;
  left: 0;
  height: 40px;
  width: 100%;
  padding-top: 5px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: var(--alert-red);
  color: var(--lt-red);
  border-bottom: 2px solid var(--lt-red);
  transition: 0.2s;
}

@media screen and (max-width: 890px) {
  .Header {
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .h-logo img {
    width: 300px;
  }

  .h-nav-group {
    height: 50px;
    width: 250px;
  }

  .btn {
    font-size: 1rem;
  }
}

@media screen and (max-width: 500px) {
  .Header {
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    padding: 60px 0;
  }

  .h-logo img {
    width: 230px;
  }

  .h-nav-group {
    height: 50px;
    width: 225px;
  }

  .btn {
    font-size: 1rem;
  }
}