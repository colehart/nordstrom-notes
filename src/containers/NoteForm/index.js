import React, { Component } from 'react';
import TextareaAutosize from 'react-autosize-textarea';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import moment from 'moment';
import { addNotes, caughtError, newNoteAdded } from '../../actions';
import './NoteForm.css';


export class NoteForm extends Component {
  constructor() {
    super()
    this.state = {
      tag: 'Work',
      text: '',
      isDisabled: true,
      wasSuccessful: false,
      timoutId: null,
    }
  }

  componentWillUnmount() {
    const { timeoutId } = this.state
    clearTimeout(timeoutId);
  }

  handleInputChange = async event => {
    const { name, value } = event.target;

    await this.setState({ [name]: value })
    if (name === 'text') this.toggleSubmit();
  }

  toggleSubmit = async () => {
    const { text } = this.state

    text
      ? await this.setState({ isDisabled: false })
      : await this.setState({ isDisabled: true })
  }

  handleSubmit = async event => {
    event.preventDefault();
    const { tag, text } = this.state
    const date = moment().format('MM-DD-YYYY')

    try {
      const successState = {
        tag: 'Work',
        text: '',
        isDisabled: true,
        wasSuccessful: true,
      }

      await this.props.addNotes(tag, text, date)
      await this.props.newNoteAdded(true)
      await this.setState(successState)
      this.resetBanner()
    }
    catch(error) {
      this.props.caughtError(error.message)
    }
  }

  resetBanner = async () => {
    const id = setTimeout(() => this.setState({ wasSuccessful: false }), 1000)
    await this.setState({ timeoutId: id })
  }

  render() {
    const { text, isDisabled, wasSuccessful } = this.state

    return (
      <section className='NoteForm'>
        <div className={ wasSuccessful ? 'nf-success' : 'hidden' }>
          Note successfully added!
        </div>
        <h2>Make Note</h2>
        <form
          className='nf-form'
          onSubmit={this.handleSubmit}
        >
          <TextareaAutosize
            className='nf-note-text'
            name='text'
            value={text}
            onChange={this.handleInputChange}
            placeholder='Write your thoughts here...'
            aria-label='Write your note here, in 250 characters or less'
            maxLength='250'
            wrap='soft'
            required
            onResize={(e) => {}}
          />
          <div className="nf-button-group">
            <div className="nf-dropdown-group">
              <label htmlFor='nf-tags'>Tag:</label>
              <select
                className="nf-dropdown nf-btn"
                name='tag'
                onChange={this.handleInputChange}
                id='nf-tags'
              >
                <option value='Work'>Work</option>
                <option value='Personal'>Personal</option>
                <option value='Hobby'>Hobby</option>
              </select>
            </div>
            <button
              className='nf-submit nf-btn'
              disabled={isDisabled}
              type='submit'
              onClick={this.handleSubmit}
            >
              Make Note
            </button>
          </div>
        </form>
      </section>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addNotes: (tag, text, date) => dispatch(addNotes(tag, text, date)),
  caughtError: (errorMessage) => dispatch(caughtError(errorMessage)),
  newNoteAdded: (bool) => dispatch(newNoteAdded(bool))
})

NoteForm.propTypes = {
  addNotes: PropTypes.func,
  caughtError: PropTypes.func,
  newNoteAdded: PropTypes.func,
}

export default connect(null, mapDispatchToProps)(NoteForm)