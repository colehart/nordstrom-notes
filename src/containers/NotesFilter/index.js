import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { caughtError } from '../../actions';
// import { filterNotes } from '../../actions';
import './NotesFilter.css';

export class NotesFilter extends Component {
  constructor () {
    super()
    this.state = {
      filter: '',
      filterParams: '',
      isDisabled: true,
      placeholderText: 'Choose a filter.',
    }
  }

  handleInputChange = async event => {
    const { name, value } = event.target;

    await this.setState({ [name]: value })
    if (name === 'filter') this.toggleParams()
  }

  toggleParams = async () => {
    const { filter } = this.state

    filter
      ? await this.setState({ isDisabled: false })
      : await this.setState({ isDisabled: true })

    this.formatPlaceholder(filter)
  }

  formatPlaceholder = async filter => {
    const dateState = { placeholderText: 'Ex: 12-18-2018' }
    const tagState = { placeholderText: 'Ex: Work, Hobby...' }

    filter === 'date'
      ? await this.setState(dateState)
      : await this.setState(tagState)
  }

  render() {
    const { filter, filterParams, isDisabled, placeholderText } = this.state;

    return (
      <section className='NotesFilter'>
        <form
          className='nl-form'
          // onSubmit={this.props.filterNotes()}
        >
          <div className='nl-filter-by'>
            <label htmlFor='filter-group'>Filter by:</label>
            <div
              className="nl-filter-group"
              id='filter-group'
              name='filter-type'
            >
              <div className='nl-filter-type'>
                <input
                  type='radio'
                  name='filter'
                  value='date'
                  id='date'
                  onChange={this.handleInputChange}
                />
                <label htmlFor="date">Date</label>
              </div>
              <div className='nl-filter-type'>
                <input
                  type='radio'
                  name='filter'
                  value='tag'
                  id='tag'
                  onChange={this.handleInputChange}
                />
                <label htmlFor='tag'>Tag</label>
              </div>
            </div>
          </div>
          <div className='nl-filter-params-group'>
            <label htmlFor='filter-params'>Search:</label>
            <input
              className='nl-filter-params'
              id='filter-params'
              name='filterParams'
              value={filterParams}
              maxLength={ filter === 'date' ? '10' : '8' }
              onChange={this.handleInputChange}
              disabled={isDisabled}
              placeholder={placeholderText}
            />
          </div>
        </form>
      </section>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  caughtError: (errorMessage) => dispatch(caughtError(errorMessage)),
  // filterNotes: () => dispatch(filterNotes()),
})

NotesFilter.propTypes = {
  caughtError: PropTypes.func,
  filterNotes: PropTypes.func,
}

export default connect(null, mapDispatchToProps)(NotesFilter)