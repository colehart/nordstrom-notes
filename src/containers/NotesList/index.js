import React from 'react';
import './NotesList.css';

export const NotesList = () => {
  return (
    <div className='NotesList'>
    </div>
  )
}