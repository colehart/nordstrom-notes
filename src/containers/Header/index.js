import React from 'react';
import './Header.css';

export const Header = () => {
  return (
    <header className='Header'>
      Header
    </header>
  )
}