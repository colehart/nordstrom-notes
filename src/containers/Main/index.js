import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { Loader } from '../../components/Loader';
import NoteForm from '../../containers/NoteForm';
import { NotesList } from '../../containers/NotesList';
import './Main.css';

export const Main = ({ isLoading, match }) => {
  return (
    <main className='Main'>
      { isLoading ? <Loader /> : '' }
      { match.path === '/'
          ? <NoteForm />
          : <NotesList />
      }
    </main>
  )
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading
})

Main.propTypes = {
  isLoading: PropTypes.bool
}

export default withRouter(connect(mapStateToProps)(Main))
