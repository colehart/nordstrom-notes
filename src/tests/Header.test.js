import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps } from '../containers/Header';
import { mockDefaultState, mockFullState } from './testMocks';

describe('Header', () => {
  describe('Header Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Header />)
    })

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it('matches error snapshot', () => {
      wrapper = shallow(<Header caughtError={'There is an error'}/>)

      expect(wrapper).toMatchSnapshot();
    })
  })

  describe('matchStateToProps', () => {
    it('should parse caughtError from state', () => {
      const expected1 = ''

      const mappedProps1 = mapStateToProps(mockDefaultState)
      expect(mappedProps1.caughtError).toEqual(expected1)

      const expected2 = 'Error: Bad API Key.'
      const mappedProps2 = mapStateToProps(mockFullState)
      expect(mappedProps2.caughtError).toEqual(expected2)
    })
  })
})
