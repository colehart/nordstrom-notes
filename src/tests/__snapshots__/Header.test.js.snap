// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Component matches error snapshot 1`] = `
<header
  className="Header"
>
  <div
    className="h-error"
  >
    There is an error
  </div>
  <Link
    aria-label="Return to the Take Note Form"
    className="h-logo"
    replace={false}
    to="/"
  >
    <img
      alt="The Nordstrom Notes logo"
      src="nordstrom-notes-logo.png"
    />
  </Link>
  <nav
    className="h-nav-group"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      aria-label="Click to return to the Take Note form"
      className="h-take-note btn"
      exact={true}
      to="/"
    >
      Make Note
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      aria-label="Click to view your notable notes"
      className="h-notable-notes btn"
      exact={true}
      to="/notes"
    >
      Notable Notes
    </NavLink>
  </nav>
</header>
`;

exports[`Header Header Component matches the snapshot 1`] = `
<header
  className="Header"
>
  <div
    className="hidden"
  />
  <Link
    aria-label="Return to the Take Note Form"
    className="h-logo"
    replace={false}
    to="/"
  >
    <img
      alt="The Nordstrom Notes logo"
      src="nordstrom-notes-logo.png"
    />
  </Link>
  <nav
    className="h-nav-group"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      aria-label="Click to return to the Take Note form"
      className="h-take-note btn"
      exact={true}
      to="/"
    >
      Make Note
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      aria-label="Click to view your notable notes"
      className="h-notable-notes btn"
      exact={true}
      to="/notes"
    >
      Notable Notes
    </NavLink>
  </nav>
</header>
`;
